### PR TITLE
Perfomance audit

### PR DESCRIPTION
**Pull request by FilbertKimasa**
In this pull request, we optimize database performance.

The screenshots below show the execution time before and after optimization.
**Before**
snap1

**After**
snap2

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -215,4 +215,6 @@ JOIN vets ON visits.vet_id = vets.id
 WHERE vets.name = 'Maisy Smith'
 GROUP BY species.name
 ORDER BY num_visits DESC
-LIMIT 1;
\ No newline at end of file
+LIMIT 1;
+
+SELECT COUNT(animal_id) FROM visits WHERE animal_id = 4;
```
Comment (Abenezer-Tilahun) [inline queries.sql:220]: - [ ] As stated in the project requirement **you were challenged with 3 queries that take too much time** (check project description for more details):
- **Use EXPLAIN ANALYZE on the previous queries to check what is happening. Take screenshots of them** .Find a way to decrease the execution time of the first query and Find a way to improve execution time of the other two queries.

- While you are making changes, check if they help by running **EXPLAIN ANALYZE. Once you succeed, take a screenshot of the EXPLAIN ANALYZE result showing that time actually decreased.**
- But In the case of yours you only check the first query so kindly make sure that you also check the remaining two query.

```
SELECT COUNT(*) FROM visits where animal_id = 4;
SELECT * FROM visits where vet_id = 2;
SELECT * FROM owners where email = 'owner_18327@mail.com'; 
```

File: schema.sql [hunk 2]
```diff
@@ -53,3 +53,7 @@ CREATE TABLE visits (
     visit_date DATE,
     PRIMARY KEY (animal_id, vet_id, visit_date)
 );
+
+ CREATE INDEX ON visits(animal_id);
+
+ ANALYZE visits;
```
Comment (Abenezer-Tilahun) [inline schema.sql]: - [ ] Good Job for **implementing indexes for the expected solution**  but kindly also make sure that you added an indexes for visits table with the `vet_id` and also for for owners table with `email` column just like the below sample.

```sql
CREATE INDEX ON visits (vet_id);
   CREATE INDEX ON owners (email);
```

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @FilbertKimasa @DevKaranJ 

Great job for implemented  the some of the Project Requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻
✔️ No Linter errors
✔️ Correct Git-Flow
✔️ Good commit messages
✔️ Professional README
✔️ Correctly Add an index for the expected solution 
✔️ Correctly check the query for the first query.

almost

## Required Changes ♻️
_Check the comments below the review._

- [x] As stated in the project requirement you are required to Run the following query to add an extra column to the owners table:
`ALTER TABLE owners ADD COLUMN email VARCHAR(120);`

- Run the following statements to add data to your database (executing them might take a few minutes): **makes sure that `INSERT INTO` queries are saved in the` data.sql` file not the` schema.sql `file**
`INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp`;

`insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';`

- [x] Kindly make sure that you included a pair of screenshots in the PR description with the results of the `EXPLAIN ANALYZE` command for each of the above queries (Screenshots should contain a total of 6 queries) for the remaining two queries .
- * `SELECT * FROM visits where vet_id = 2;`
* `SELECT * FROM owners where email = 'owner_18327@mail.com';`

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — COMMENTED** [review]
# Hi @FilbertKimasa & @DevKaranJ 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- It seems your review request is appearing as a first review although it's been reviewed previously to avoid any confusion I'd suggest the student who first submitted the PR resubmit again.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Pull request merged**